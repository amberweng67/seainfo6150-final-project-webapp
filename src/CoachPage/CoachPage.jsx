import React from 'react';
import styles from './CoachPage.module.css';
import useCoach from '../Hooks/useCoach';

const CoachPage = (props) => { 
  const {match, coachList, setCoachList, user} = props;
  const [coach] = useCoach(match.params.coachId);
  if (!coach) {
    return null;
  }

  const addToCoachList = () => {
    setCoachList([...coachList, coach]);
  };

  const isAddToCoachListDisabled = coachList.find(c => c.id === coach.id);

  return (
    <div className={styles.container}>
      <div className={styles.imgContainer}>
        <img 
        alt="Coach profile" src={require(`../assets/${coach.id}.webp`)} />
      </div>
      <div>
        <h1>Hey, I'm {coach.name}</h1>
        <div>
          <b>{coach.title}</b> at <b>{coach.company}</b>
        </div>
        <h3>About Me</h3>
        <p>
        {coach.intro}
        </p>
        <div>
        {user && coachList && <button onClick={addToCoachList} 
        className={isAddToCoachListDisabled ? styles.disabledButton : ''} disabled={isAddToCoachListDisabled}>Add to my list</button>}
        </div>
      </div>
    </div>
  )
}

export default CoachPage;
