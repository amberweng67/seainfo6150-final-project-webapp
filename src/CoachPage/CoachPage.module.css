.container {
  padding: 64px;
  display: flex;
}

.imgContainer > img {
  object-fit: cover;
  height: 600px;
  width: 400px;
  margin-right: 64px;
}

.disabledButton {
  background-color: rgba(0, 0, 0, 0.2);
}

.disabledButton:hover {
  opacity: 1;
  cursor: not-allowed;
}

@media screen and (max-width: 600px) {
  .imgContainer > img {
    height: 200px;
    width: 100px;
  }
}

@media screen and (max-width: 960px) and (min-width:601px) {
  .imgContainer > img {
    height: 400px;
    width: 200px;
  }
}