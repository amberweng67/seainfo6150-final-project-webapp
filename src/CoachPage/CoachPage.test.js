import React from "react";
import CoachPage from "./CoachPage.jsx";

it("renders the coach page component correctly", () => {
  const { container } = render(<CoachPage match={{params: 'test'}}/>);
  expect(container).toMatchSnapshot();
});
