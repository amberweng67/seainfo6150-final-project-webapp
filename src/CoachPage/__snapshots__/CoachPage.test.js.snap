// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the coach page component correctly 1`] = `<div />`;
